keps/sig-cluster-lifecycle/kubeadm-config-ha-support.md

```diff
@@ -0,0 +1,147 @@
+---
+kep-number: TBD
+title: Augment Kubeadm Config to Enable Upgrades of HA Clusters
+authors:
+  - "@mattkelly"
+owning-sig: sig-cluster-lifecycle
+reviewers:
+  - TBD
+approvers:
+  - TBD
+editor:
+  - "@mattkelly"
+creation-date: 2018-03-08
+last-updated: 2018-03-09
+status: provisional
+see-also:
+  - [KEP kubeadm join --master workflow](https://github.com/kubernetes/community/pull/1707) (in progress)
+---
+
+# Augment Kubeadm Config to Enable Upgrades of HA Clusters
+
+## Table of Contents
+
+* [Augment Kubeadm Config to Enable Upgrades of HA Clusters](#augment-kubeadm-config-to-enable-upgrades-of-ha-clusters)
+  * [Table of Contents](#table-of-contents)
+  * [Summary](#summary)
+  * [Motivation](#motivation)
+     * [Goals](#goals)
+     * [Non-Goals](#non-goals)
+     * [Challenges and Open Questions](#challenges-and-open-questions)
+  * [Proposal](#proposal)
+     * [Implementation Details](#implementation-details)
+        * [Understanding the Situation Today](#understanding-the-situation-today)
+        * [Adding Additional Master-Specific ConfigMaps](#adding-additional-master-specific-configmaps)
+           * [Key Design Considerations and Benefits](#key-design-considerations-and-benefits)
+              * [Parallel Node Creation](#parallel-node-creation)
+              * [Guaranteed Consistent kubeadm-config](#guaranteed-consistent-kubeadm-config)
+     * [Risks and Mitigations](#risks-and-mitigations)
+  * [Implementation History](#implementation-history)
+  * [Drawbacks](#drawbacks)
+  * [Alternatives](#alternatives)
+
+## Summary
+
+Currently, `kubeadm upgrade` of a master in a multi-master cluster will fail without workarounds because of a lack of node-specific master configuration in the `kubeadm-config` ConfigMap that is created at `init` time and later referenced during an upgrade.
```

Comment:
IMO it is a little bit unfair saying that the current implementation will fail (it hints a wrong implementation)
What about rephrasing this a little bit and telling that the kubeadm-config-machineUID was originally designed for supporting upgrades in single-master clusters, and now, As we move towards supporting HA natively in kubeadm, this design should be improved?

Reply:
I agree that this could be misinterpreted - I will reword it along the lines of your suggestion.